### PR TITLE
fix(md-menu): allow nested md-menu-content again

### PR DESCRIPTION
**Pull request by IMM0rtalis**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
Fixes misbehavior from #10198

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #10540

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by IMM0rtalis** [comment]
@Splaktar Somehow the test seems to fail because of a different reason than my code. Do you know what happened here?

File: src/components/menu/js/menuDirective.js [hunk 1]
```diff
@@ -198,8 +196,8 @@ function MenuDirective($mdUtil) {
       throw Error(INVALID_PREFIX + 'Expected the menu to have a trigger element.');
     }
 
-    if (!contentEl || contentEl.nodeName !== 'MD-MENU-CONTENT') {
-      throw Error(INVALID_PREFIX + 'Expected the menu to contain a `md-menu-content` element.');
+    if (templateElement.children().length !== 2) {
+      throw Error(INVALID_PREFIX + 'Expected two children elements.');
```
Comment (DevVersion) [inline src/components/menu/js/menuDirective.js]: It would be nice to still kind of refer to the `md-menu-content` element.

**Comment by IMM0rtalis** [comment]
@Splaktar can this be added to milestone/release 1.1.7 or is it too late?

**Comment by Splaktar** [comment]
@IMM0rtalis there wasn't time to get this into 1.1.7 due to how long of a process it is to get through presubmits. Please squash your commits and then I'll send this off for presubmit testing.

**Comment by IMM0rtalis** [comment]
@Splaktar done.
Yeah, I thought so when I saw 1.1.7 released ;)

**Comment by Splaktar** [comment]
@IMM0rtalis in order to improve the generated changelog and make sure that the related issue gets closed, please include `Fixes #10540` on the last line of your commit message in the future. More details are in the [commit message guidelines](https://github.com/angular/material/blob/master/.github/CONTRIBUTING.md#-commit-message-format).

**Pull request merged**